boards/posix/nrf52_bsim/cmsis_compiler.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * This header defines replacements for CMSIS compiler attributes.
+ */
+
+#ifndef CMSIS_COMPILER_H_
```

Comment:
Better use the whole path in the header guard please (both due to convention, and in case somebody want to do trickery with multiple inclusions on different cmsis_compiler.h)